src/main/java/de/epiceric/shopchest/ShopCommand.java

```diff
@@ -421,65 +423,77 @@ private void create(String[] args, ShopType shopType, Player p) {
+        plugin.debug(player.getName() + "'s prices are higher than the minimum");
 
         if (sellEnabled && buyEnabled) {
             if (plugin.getShopChestConfig().buy_greater_or_equal_sell) {
                 if (buyPrice < sellPrice) {
-                    p.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.BUY_PRICE_TOO_LOW, new LocalizedMessage.ReplacedRegex(Regex.MIN_PRICE, String.valueOf(sellPrice))));
+                    player.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.BUY_PRICE_TOO_LOW, new LocalizedMessage.ReplacedRegex(Regex.MIN_PRICE, String.valueOf(sellPrice))));
                     return;
                 }
             }
         }
 
-        plugin.debug(p.getName() + "'s buy price is high enough");
+        plugin.debug(player.getName() + "'s buy price is high enough");
 
-        ItemStack itemStack = new ItemStack(Utils.getPreferredItemInHand(p).getType(), amount, Utils.getPreferredItemInHand(p).getDurability());
-        itemStack.setItemMeta(Utils.getPreferredItemInHand(p).getItemMeta());
+        ItemStack itemStack = new ItemStack(Utils.getPreferredItemInHand(player).getType(), amount, Utils.getPreferredItemInHand(player).getDurability());
+        itemStack.setItemMeta(Utils.getPreferredItemInHand(player).getItemMeta());
 
         if (Enchantment.DURABILITY.canEnchantItem(itemStack)) {
             if (itemStack.getDurability() > 0 && !plugin.getShopChestConfig().allow_broken_items) {
-                p.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.CANNOT_SELL_BROKEN_ITEM));
+                player.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.CANNOT_SELL_BROKEN_ITEM));
                 return;
             }
         }
 
-        plugin.debug(p.getName() + "'s item is not broken (or broken items are allowed through config)");
+        plugin.debug(player.getName() + "'s item is not broken (or broken items are allowed through config)");
 
         double creationPrice = (shopType == ShopType.NORMAL) ? plugin.getShopChestConfig().shop_creation_price_normal : plugin.getShopChestConfig().shop_creation_price_admin;
         if (creationPrice > 0) {
-            if (plugin.getEconomy().getBalance(p) < creationPrice) {
-                p.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.SHOP_CREATE_NOT_ENOUGH_MONEY, new LocalizedMessage.ReplacedRegex(Regex.CREATION_PRICE, String.valueOf(creationPrice))));
+            if (plugin.getEconomy().getBalance(player) < creationPrice) {
+                player.sendMessage(LanguageUtils.getMessage(LocalizedMessage.Message.SHOP_CREATE_NOT_ENOUGH_MONEY, new LocalizedMessage.ReplacedRegex(Regex.CREATION_PRICE, String.valueOf(creationPrice))));
                 return;
             }
         }
 
-        plugin.debug(p.getName() + " can pay the creation price");
+        plugin.debug(player.getName() + " can pay the creation price");
+
+        if (plugin.hasPlotSquared()) {
```

Comment:
I think this should be in the interact event, since the player could stand in his own plot to enter the command, but click on a chest in another plot.